waiter/src/waiter/scheduler.clj

```diff
@@ -486,16 +489,29 @@
                   instance-id->failed-health-check-count' (pc/map-from-keys #((fnil inc 0) (get instance-id->failed-health-check-count %))
                                                                             (keys instance-id->unhealthy-instance'))]
               (metrics/reset-counter
-                (metrics/service-counter id "instance-counts" "failed")
+                (metrics/service-counter service-id "instance-counts" "failed")
                 (count all-failed-instances))
-              (recur (conj service-id->health-check-context'
-                           {id {:instance-id->unhealthy-instance instance-id->unhealthy-instance'
-                                :instance-id->tracked-failed-instance instance-id->tracked-failed-instance'
-                                :instance-id->failed-health-check-count instance-id->failed-health-check-count'}})
-                     scheduler-messages' remaining))
-            (do (log/info (timing-message-fn) "for" (count service->service-instances) "services.")
-                {:service-id->health-check-context service-id->health-check-context'
-                 :scheduler-messages scheduler-messages}))))
+              (recur
+                (assoc service-id->health-check-context'
+                  service-id {:instance-id->unhealthy-instance instance-id->unhealthy-instance'
+                              :instance-id->tracked-failed-instance instance-id->tracked-failed-instance'
+                              :instance-id->failed-health-check-count instance-id->failed-health-check-count'})
+                (cond-> healthy-service-ids
+                        (seq healthy-instances) (conj service-id))
+                scheduler-messages'
+                remaining))
+            (let [healthy-service-ids (->> scheduler-messages
```

Comment:
Remove this threading macro

Reply:
Removed. Actually, this snippet was not even needed. It also triggered my refactor of `available-service-ids` to a set.